packages/core/services/HttpServiceBase/index.ts

```diff
@@ -210,6 +210,30 @@ export default class HttpServiceBase {
         return new RestServiceResponse(response.data);
     }
 
+    public async put<T>(url: string, body: string): Promise<RestServiceResponse<T>> {
+        const encodedUrl = HttpServiceBase.encodeURI(url);
+        const config = { headers: { "Content-Type": "application/json" } };
+
+        let response;
+        try {
+            // if this fails, bubble up exception
+            response = await retry.execute(() => this.httpClient.put(encodedUrl, body, config));
+        } catch (err) {
+            // Specific errors about the failure from services will be in this path
+            if (axios.isAxiosError(err) && err?.response?.data?.message) {
+                throw new Error(JSON.stringify(err.response.data.message));
+            }
+            throw err;
+        }
+
+        if (response.status >= 400 || response.data === undefined) {
+            // by default axios will reject if does not satisfy: status >= 200 && status < 300
```

Comment:
if the default already rejects why is this a condition?

Reply:
This is a pattern used by all the http service methods that made sense to me before but which I'm now having trouble defending. My interpretation is that when axios rejects a call, it does not necessarily throw a javascript error object and may bypass the catch, so the status must be checked if we want it to throw something